products/bridge/test/test.ts

```diff
@@ -139,20 +148,24 @@ describe("Bridge", function () {
       success,
       result,
       nonce,
-      signerIndices,
       signatures
     );
 
-    const blockNum = await ethers.provider.getBlockNumber();
-
     const filter = twin1.filters.Succeeded();
-    const logs = await twin1.queryFilter(filter, blockNum - 100, blockNum);
+    const logs = await twin1.queryFilter(filter);
```

Comment:
As discussed, it's essential to only consider events in finalized blocks, therefore the code before refactoring fetched `ethers.provider.getBlockNumber('finalized')` and queried only until that block number.